Fix scaffolded files and add sanity checks to github workflow

mollux opened the pull request:
<!-- ## Which branch should I use for my PR?

Assuming that:

a = current major release
b = current minor release
c = future major release

* a.x for any features and enhancements (e.g. 4.x)
* a.b for any bug fixes (e.g. 4.0, 4.1, 4.2)
* c.x for any features, enhancements or bug fixes with backward compatibility breaking changes (e.g. 5.x) -->

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] <!-- All PRs must maintain or improve code coverage -->
| Related user documentation PR URL      | mautic/mautic-documentation#... <!-- required for new features -->
| Related developer documentation PR URL | mautic/developer-documentation#... <!-- required for developer-facing changes -->
| Issue(s) addressed                     | Fixes #... <!-- prefix each issue number with "Fixes #", no need to create an issue if none exists, explain below instead -->

<!--
Additionally (see https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#work-on-your-pull-request):
 - Always add tests and ensure they pass.
 - Bug fixes must be submitted against the lowest maintained branch where they apply
   (lowest branches are regularly merged to upper ones so they get the fixes too.)
 - Features and deprecations must be submitted against the "4.x" branch.
-->

#### Description:

This PR addresses steps in the QA process that are easily forgotten:
- the `phpstan-baseline.neon` should only shrink, not altered (see [the commits after Nov 9, 2021](https://github.com/mautic/mautic/commits/4.x/phpstan-baseline.neon) for recent changes)
- the files used for scaffolding should be in sync with the original ones (e.g. the [original `.gitignore`](https://github.com/mautic/mautic/commits/4.x/.gitignore) has commits not present in the  [scaffolded `.gitignore`](https://github.com/mautic/mautic/commits/4.x/app/assets/scaffold/files/example.gitignore))

<!--
Please write a short README for your feature/bugfix. This will help people understand your PR and what it aims to do. If you are fixing a bug and if there is no linked issue already, please provide steps to reproduce the issue here.
-->

#### Steps to test this PR:

The tests in this PR fail intentionally to show the effect:
see https://github.com/mautic/mautic/pull/10857/checks
* the [`scaffolded files mismatch`](https://github.com/mautic/mautic/runs/5172853079?check_suite_focus=true#logs) fails when the scaffolded files that are not in sync with the original ones
* the [`PHPStan baseline changes`](https://github.com/mautic/mautic/runs/5172853097?check_suite_focus=true#logs) fails when there are unwanted changes to the `phpstan-baseline.neon` file

Once this makes sense to the reviewers of this PR, following steps are needed **before merging**:
* fix the scaffolded files
* revert the commit(s) that alter the `phpstan-baseline.neon` files


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10857"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

mollux committed:
add extra checks in the misc job

escopecz commented on .github/workflows/tests.yml:
```diff
@@ -152,6 +152,27 @@ jobs:
+          rm diff_commands.sh
+          if [[ $(wc -l </tmp/diff_command_output.txt) -ge 1 ]]; then
+            echo "some scaffolded files were not updated"
+            echo "Please apply the same changes in the files mentioned above"
```
Is there some command the dev can run to apply the changes? Or tell them somehow int he error message how to fix the problem?

mollux replied:
the files are mentioned in the output
(example of the current situation)

```
Files ./package-lock.json and app/assets/scaffold/files/package-lock.json differ
Files ./.gitignore and app/assets/scaffold/files/example.gitignore differ
Files ./upgrade.php and app/assets/scaffold/files/upgrade.php differ
```

I could add some line like 
```
please ensure the changes you made in the file(s) mentioned above are also present in their linked file in the `app/assets/scaffold/files` folder
```
I could parse that output and make example `cp` commands, but that's a bit too far IMO.
As some people only use an IDE and not CLI

escopecz replied:
Let's step back. I missed the reason behind the app/assets/scaffold folder, so I'm the great example of a person who has no idea what's going on. I don't have to play dumb but just myself :)

Why do we have 3 files that must be the same in 2 different locations? Can it be done with symlinks?

mollux replied:
Those files are needed for the composer based flow for Mautic, as they live outside of `mautic/core-lib`  
I assume the decision for copies of the files was discussed and made when setting up the composer flow.  

e.g. Drupal has the same issue: both tarball based installs and composer based installs are supported.  
But the composer based install can't rely on the files around the application (e.g. .htaccess, index.php)

so those files are changes twice, in their normal location, and in a location that can be used for scaffolding.
See e.g. [this commit in Drupal core](https://git.drupalcode.org/project/drupal/-/commit/d0be0af780017fa7af8471a5c93c7c1aa4a114a7): 
it changes the .htaccess file in 2 locations.

there is an [ongoing issue about this in Drupal core](https://www.drupal.org/project/drupal/issues/3075954) to remove the duplicate ones, and only rely on the scaffolded folder

escopecz replied:
I understand now. Thanks! The scaffolding check makes sense 👍 

I'll approve this PR once the CI is green.

mollux replied:
all tests green :)

mollux committed:
Fix scaffolded files that are out of sync

escopecz reviewed (dismissed):
Thanks! This is very helpful 👍

kuzmany commented:
This also stop upgrade by `upgrade.php`? I do not see any mention in description , but see changelog

mollux commented:
@kuzmany the description only mentions 1 example, but there are currently 3 files not in sync that are fixed in this PR.  
See [a previous build](https://github.com/mautic/mautic/runs/5172853079?check_suite_focus=true) for the 3 files that failed.

The test obviously passes now, so it's not easy to see indeed :)

escopecz commented:
@kuzmany the changes here are just to sync the same files in the root dir with the same files in the `app/assets/scaffold/files` dir. So the changes in those file were already done in previous PRs but weren't copied to the scaffold.

kuzmany commented:
@mollux this is something in CI status what we you need check 

![image](https://user-images.githubusercontent.com/462477/154134930-54a371ed-60b3-4b00-8c54-5ed614c25e34.png)

mollux committed:
reverted renamed tests, as it breaks the reporting

mollux commented:
@kuzmany aha, I renamed 2 tests, but that's a bad idea, as those names are in the repo configuration

I reverted that in a54f3ca6c41c162152dc329654c6a6270a320d44

kuzmany committed:
Merge branch '4.x' into sanity-checks
